include/openthread/dataset.h

```diff
@@ -605,4 +606,27 @@ otError otDatasetUpdateTlvs(const otOperationalDataset *aDataset, otOperationalD
 } // extern "C"
 #endif
 
+#ifdef __cplusplus
+/**
+ * Defines equality for two Timestamps.
+ */
+inline bool operator==(const otTimestamp &aLhs, const otTimestamp &aRhs)
+{
+    return (aLhs.mSeconds == aRhs.mSeconds) && (aLhs.mTicks == aRhs.mTicks) &&
+           (aLhs.mAuthoritative == aRhs.mAuthoritative);
+}
```

Comment:
Two issues here:
- OpenThread public APIs MUST be C. We cannot and should not use C++. The header file name is `dataset.h`. 
- No inline implementation in C header for any public/platform APIs.

If you want these, you should define C APIs for them (see examples like `otIp6ArePrefixesEqual()`).